File: src/frontend/src/app/components/atendimento/convenio-categoria/model/convenio-categoria-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class ConvenioCategoriaDTO {
  @Expose() id?: string;
  @Expose() convenioId?: string;
  @Expose() convenioNome?: string;
  @Expose() nome?: string;
  @Expose() codigoAnsPlano?: string;
  @Expose() ativo?: boolean;
  @Expose() createdAt?: Date;
  @Expose() updatedAt?: Date;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/conveniocategoria/dto/ConvenioCategoriaDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.conveniocategoria.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class ConvenioCategoriaDTO implements DTO {

    private UUID id;

    private UUID convenioId;
    private String convenioNome;

    private String nome;
    private String codigoAnsPlano;
    private Boolean ativo;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
